### DISCUSSION TITLE
VSCode keyboard shortcut "GitHub Copilot: Enable/Disable Copilot completions" does not work in editor

### DISCUSSION THREAD

**Discussion by jaakko-paavola** [community/community#79048] [Copilot Conversations]
### Select Topic Area

Bug

### Body

In VSCode, as the Copilot completions being turned on causes significant slowing on my system, I would like to be able to turn them off with a keyboard shortcut whenever I don't need them.

I have set
{"key": "ctrl+shift+meta+c",   "command": "github.copilot.toggleCopilot" },
and also tried various other key combinations and e.g. adding "when": "editorFocus", but I cannot get it to disable/enable the completions when the editor has the focus. However, when the focus is in Keyboard Shortcuts tab, then the shortcut seemingly works (I can see the small Copilot icon in the bottom right corner getting the diagonal strike over it), but when I change the focus to editor, the completions are still on (and the diagonal strike over the icon is gone).

Here's my setup:
Version: 1.86.0-insider
Commit: f87fb292c3af22e51a7f5c628496cd8d6fd9ecd5
Date: 2023-12-13T08:07:26.768Z
Electron: 27.1.3
ElectronBuildId: 25612240
Chromium: 118.0.5993.159
Node.js: 18.17.1
V8: 11.8.172.18-electron.0
OS: Linux x64 6.2.0-39-generic

and GitHub Copilot v1.143.0

**Comment by BezRegi** [comment]
> ### Select Topic Area
> 
> Bug
> 
> ### Body
> 
> In VSCode, as the Copilot completions being turned on causes significant slowing on my system, I would like to be able to turn them off with a keyboard shortcut whenever I don't need them.
> 
> I have set
> {"key": "ctrl+shift+meta+c",   "command": "github.copilot.toggleCopilot" },
> and also tried various other key combinations and e.g. adding "when": "editorFocus", but I cannot get it to disable/enable the completions when the editor has the focus. However, when the focus is in Keyboard Shortcuts tab, then the shortcut seemingly works (I can see the small Copilot icon in the bottom right corner getting the diagonal strike over it), but when I change the focus to editor, the completions are still on (and the diagonal strike over the icon is gone).
> 
> Here's my setup:
> Version: 1.86.0-insider
> Commit: f87fb292c3af22e51a7f5c628496cd8d6fd9ecd5
> Date: 2023-12-13T08:07:26.768Z
> Electron: 27.1.3
> ElectronBuildId: 25612240
> Chromium: 118.0.5993.159
> Node.js: 18.17.1
> V8: 11.8.172.18-electron.0
> OS: Linux x64 6.2.0-39-generic
> 
> and GitHub Copilot v1.143.0

**Comment by jaakko-paavola** [comment]
This seems to be a problem in tabs where the language mode is Python. If I trigger the shortcut and then see the Copilot status menu, the line that says Enable/Disable Completion for 'python' never changes its status, only the "general" Disable/Enable Completions above it does. If the language mode is e.g. JSON with comments, then the shortcut works as expected: both the Enable/Disable Completion for 'jsonc' and the "general" Disable/Enable Completions change their statuses from the key shortcut and the Copilot icon gets struck over.

**Comment by DanteMarshal** [comment]
I'm facing the same issue, and I can confirm that even calling it using the command palette doesn't work.
Could you find any workaround ?
